packages/puppeteer-core/src/bidi/core/UserContext.ts

```diff
@@ -115,11 +123,15 @@ export class UserContext extends EventEmitter<{
     return browsingContext;
   }
 
-  async close(): Promise<void> {
-    const promises = [];
-    for (const browsingContext of this.#browsingContexts.values()) {
-      promises.push(browsingContext.close());
-    }
-    await Promise.all(promises);
+  async remove(): Promise<void> {
+    // TODO: implement incognito context https://github.com/w3c/webdriver-bidi/issues/289.
+    // TODO: Uncomment once we have adopted the new browsing contexts.
+    // const promises = [];
+    // for (const browsingContext of this.#browsingContexts.values()) {
```

Comment:
this should be handled by bidi, I think we do not need to close explicitly but we might need to wait till it is destroyed.

Reply:
That's good then. We can just fix the logic here since Puppeteer does this for us and remove the logic in Puppeteer once we implement this in Bidi

Reply:
Actually, we can just remove this and call removeUserContext I guess. I assume that command doesn't resolve until all browsing contexts are deleted anyway.